[MRG + 2] [MAINT] Update to Sphinx-Gallery 0.1.7

Titan-C opened the pull request:
latest release of Sphinx-Gallery

Titan-C committed:
[MAINT] Update to Sphinx-Gallery 0.1.6

jnothman commented:
Could you please point to a change-log? Thanks

amueller commented:
merge if tests pass, I guess? Not sure what should be reviewed? Can you add an entry to whatsnew?

lesteve commented:
> Not sure what should be reviewed?

I guess having a glance at the gallery and make sure nothing looks off is a good start: https://7552-843222-gh.circle-artifacts.com/0/home/ubuntu/scikit-learn/doc/_build/html/stable/auto_examples/index.html

Titan-C commented:
The relevant changes of Sphinx-gallery for Scikit-learn from version 0.1.4 to 0.1.6 are
v0.1.6

Bug Fixes

- Sphinx-Gallery now raises an exception if the matplotlib bakend can not be set to 'agg'.
- Fix backreferences.identify_names when module is used without attribute
- Raise FileNotFoundError when README.txt is not present in the main directory of the examples gallery(#164). Also include extra empty lines after reading README.txt to obtain the correct rendering of the html file.(#165)

- CSS. Now a tooltip is displayed on the source code blocks to make the doc-resolv functionality more discorverable. Function calls in the source code blocks are hyperlinks to their online documentation.
Download buttons have a nicer look across all themes offered by Sphinx

Titan-C commented:
Ok this one looks a bit ugly a CSS conflict. The tooltip for links to functions documentation appears below the sidebar to hide the left menu.
![code_tooltip](https://cloud.githubusercontent.com/assets/713451/20920370/ff9bc196-bb9f-11e6-9ba1-0d2ff079fd07.png)

Titan-C committed:
[CSS] Tooltip change codelinks

Tooltip starts with same left alignment as link. This is to avoid
tooltip appearing below menu hide sidebar button. The z-index is not
working as should putting it on top of the sidebar.

Titan-C commented:
I pushed a fix for the tooltip alignment. Now it left aligns to with the world so that is not anymore overlapping with the sidebar button. For reasons that go beyond my understanding on CSS raising the z-index of the tooltip did not help to put it on top of the sidebar. The highest index is from the bootstrap themes and is 1051, but even putting the tooltip to 2000 did not change anything.

GaelVaroquaux commented on doc/sphinxext/sphinx_gallery/_static/gallery.css:
```diff
@@ -209,7 +209,6 @@ a.sphx-glr-code-links[tooltip]:hover:before{
     text-align: center;
     text-indent: 0;
-    margin-left: -5em;
+    margin-left: 0; /* Has to be zero to avoid overlapping with sidebar */
```
Is this change a backport of something in sphinx_gallery, or is it a modification specific for scikit-learn?

I am asking because I am worried that if it is specific to scikit-learn, next time that we upgrade, we will break something.

Titan-C replied:
It is a modification done in place for now for scikit learn. Which I believe to be reasonable enough to  transfer to sphinx-gallery.
Code tooltips are a recent feature in Sphinx-Gallery from your last PR as you wanted to make their use more explicit for Scipy-lectures code examples.
From a personal point of view, I see little advantage on trying to make them look a bit centered relative to the link and left aligning seems good enough. I was once slightly annoyed  they apear bellow the link, because if 2 consecutive lines have function calls, the first tooltip will block the second link and you have to move the mouse around to make them disappear and select the second function link. Then I thought making them inline would be better, and then again I thought that a nested function call I would experience the same issue, so I let it be.

GaelVaroquaux replied:
OK. I'll have a look once CircleCI is finished running, and we should either consider porting this back to sphinx gallery, or making the CSS change in a scikit-learn specific file.

GaelVaroquaux replied:
OK, looks good. +1 for backport into sphinx-gallery

amueller commented:
LGTM

jnothman commented:
@Titan-C: for future reference, we have shortcuts to Circle CI artifacts like http://scikit-learn.org/circle?7560/auto_examples

I'm not entirely happy with those tooltips. I'm okay to merge this now, despite, but I'd rather:
* "Link to" -> "View"
* Using `<a title="...">` rather than some custom intrusive tooltip.

raghavrv commented:
> I guess having a glance at the gallery and make sure nothing looks off is a good start: https://7552-843222-gh.circle-artifacts.com/0/home/ubuntu/scikit-learn/doc/_build/html/stable/auto_examples/index.html

I checked for around 20 examples, it looks fine to me...

amueller commented:
Do we want to wait with this until the sphinx 1.5 bug is fixed? https://github.com/sphinx-gallery/sphinx-gallery/pull/137

lesteve commented:
> Do we want to wait with this until the sphinx 1.5 bug is fixed? sphinx-gallery/sphinx-gallery#137

I would say yes.

Titan-C commented:
I'm keeping track on what shall get in the next Sphinx-Gallery bug fix release apart from bug fix to Sphinx v1.5 new changes.

Concerning code tooltips @jnothman 
>I'm not entirely happy with those tooltips. I'm okay to merge this now, despite, but I'd rather:
> * "Link to" -> "View"
> * Using <a title="..."> rather than some custom intrusive tooltip.

* I can go for the change "Link to" -> "View"
* Using <a title="..."> is not good. We already experienced problems with using title in read the docs theme. (See https://github.com/sphinx-gallery/sphinx-gallery/pull/145#issuecomment-247550667)

jnothman commented:
Thanks @Titan-C, I don't really understand why title is breaking with readthedocs, or why that should not be considered a bug on the part of readthedocs (title is, of course, the correct way to do this in Semantic HTML)

Titan-C commented:
Title does not only break in readthedocs. It breaks on all themes. I retested it last night, the reason is the browser itself will popup the info box with the style defined by the browser. When @GaelVaroquaux tried to put CSS style to this box, the styled box appears first and then a little later the browser pops again its own box. For that reason there is duplication.
There is of course the option of not to style this at all and let the browser handle the title attribute. 
On the same thread of issue https://github.com/sphinx-gallery/sphinx-gallery/pull/145#issuecomment-247558431, Gaël references that title has no effect on web readers for blind people so there is no true advantage for the title attribute.

jnothman commented:
Oh yes, I intended to let the browser handle it. I find the CSS popup
unnecessarily intrusive, personally.

title may not be used in web readers. In the past I think it has been used
by search engines, but that's probable not relevant to our application.

On 12 December 2016 at 20:50, Óscar Nájera <notifications@github.com> wrote:

> Title does not only break in readthedocs. It breaks on all themes. I
> retested it last night, the reason is the browser itself will popup the
> info box with the style defined by the browser. When @GaelVaroquaux
> <https://github.com/GaelVaroquaux> tried to put CSS style to this box,
> the styled box appears first and then a little later the browser pops again
> its own box. For that reason there is duplication.
> There is of course the option of not to style this at all and let the
> browser handle the title attribute.
> On the same thread of issue sphinx-gallery/sphinx-gallery#145 (comment)
> <https://github.com/sphinx-gallery/sphinx-gallery/pull/145#issuecomment-247558431>,
> Gaël references that title has no effect on web readers for blind people so
> there is no true advantage for the title attribute.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7986#issuecomment-266385514>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz6_zeFCJ6b0vIjVSk4I2N9UELzc6Xks5rHRkBgaJpZM4LE3zi>
> .
>

Titan-C committed:
[MAINT] Update to Sphinx-Gallery 0.1.7

Titan-C committed:
[doc build] Latest version of sphinx

lesteve commented:
This does not appear to be an umodified sphinx-gallery version. Running copy_sphinxgallery.sh and diffing I get:

```diff
diff --git a/doc/sphinxext/sphinx_gallery/_static/gallery.css b/doc/sphinxext/sphinx_gallery/_static/gallery.css
index a7e2890..9de9b97 100644
--- a/doc/sphinxext/sphinx_gallery/_static/gallery.css
+++ b/doc/sphinxext/sphinx_gallery/_static/gallery.css
@@ -195,11 +195,11 @@ a.sphx-glr-code-links:hover{
     text-decoration: none;
 }
 
-a.sphx-glr-code-links[tooltip]:hover:before{
+a.sphx-glr-code-links[title]:hover:before{
     background: rgba(0,0,0,.8);
     border-radius: 5px;
     color: white;
-    content: attr(tooltip);
+    content: attr(title);
     padding: 5px 15px;
     position: absolute;
     z-index: 98;
diff --git a/doc/sphinxext/sphinx_gallery/docs_resolv.py b/doc/sphinxext/sphinx_gallery/docs_resolv.py
index b2757d2..12346b5 100644
--- a/doc/sphinxext/sphinx_gallery/docs_resolv.py
+++ b/doc/sphinxext/sphinx_gallery/docs_resolv.py
@@ -358,7 +358,7 @@ def _embed_code_links(app, gallery_conf, gallery_dir):
 
     # patterns for replacement
     link_pattern = ('<a href="%s" class="sphx-glr-code-links" '
-       'tooltip="Link to documentation for %s">%s</a>')
+                    'title="Link to documentation for %s">%s</a>')
     orig_pattern = '<span class="n">%s</span>'
     period = '<span class="o">.</span>'

```

lesteve commented:
Also can you have `[doc build]` somewhere in your commit message in order to force generation of the full html documentation?

Titan-C commented:
> This does not appear to be an umodified 0.1.7 sphinx-gallery version, can you fix this? Running copy_sphinxgallery.sh and diffing I get
Indeed doing this update I noticed this. Version 0.1.7 has still tooltip and it never goes to title. But I was testing with style at that time too. For some reason I pushed to PyPI a version with title, that is wrong. I'm currently looking how to change the file in PyPI

lesteve commented:
> For some reason I pushed to PyPI a version with title, that is wrong. I'm currently looking how to change the file in PyPI

I am pretty sure you can not do that. Once the release is done on PyPI you can add more files but not change the files that have already been uploaded.

lesteve commented:
Also can you add an entry in whats_new?

Titan-C commented:
> Also can you add an entry in whats_new?

Do I put it in bug fixes? Shall I list all the change from sphinx-gallery 0.1.4->0.1.7 or say that this is mostly on the Sphinx-build Issue.

> I am pretty sure you can not do that. Once the release is done on PyPI you can add more files but not change the files that have already been uploaded.

Yes I'm required to have a new version. I'll test first in test PyPI to see how the ordering goes

lesteve commented:
> Do I put it in bug fixes? Shall I list all the change from sphinx-gallery 0.1.4->0.1.7 or say that this is mostly on the Sphinx-build Issue.

Maybe in the enhancements section. Just stay brief and mention that sphinx-gallery has been updated to 0.1.7 and give the PR and your username (look at other entries how to do it with the `issue` and `user` sphinx directive).

lesteve commented:
> Yes I'm required to have a new version. I'll test first in test PyPI to see how the ordering goes

Unless the code you uploaded for 0.1.7 has a big regression, I would not bother to much about it since 0.1.8 is going to be released in not such a long time IIRC.

Titan-C committed:
Whats new update Sphinx-Gallery 0.1.7

Titan-C commented:
> Unless the code you uploaded for 0.1.7 has a big regression, I would not bother too much about it since 0.1.8 is going to be released in not such a long time IIRC.

the PyPI version of 0.1.7 uses the title CSS atribute. No one has complained, but that gives this problem https://github.com/sphinx-gallery/sphinx-gallery/pull/145#issuecomment-247550667.

Yes 0.1.8 shall be out as soon as the relative path PR https://github.com/sphinx-gallery/sphinx-gallery/pull/190 is approved. Sphinx-Gallery 0.1.8 would contain the style changes requested here, of dropping the tooltip in favour of title and letting the browser handle the style of it.

lesteve commented:
HTML doc for the CircleCI build can be found [here](https://8292-843222-gh.circle-artifacts.com/0/home/ubuntu/scikit-learn/doc/_build/html/stable/auto_examples/index.html). I looked at the gallery and I can't find any obvious issues.

jnothman commented:
Are the download buttons meant to be side-by-side like on sphinx-gallery.readthedocs.io? They're appearing one above the other.

lesteve commented:
> Are the download buttons meant to be side-by-side like on sphinx-gallery.readthedocs.io? They're appearing one above the other.

Hmmm not sure to be perfectly honest.

I would still be in favour of merging this PR to be able to use sphinx 1.5 and have CircleCI running fine. We can sort out cosmetic in a later update of sphinx-gallery.

GaelVaroquaux commented:
>     Are the download buttons meant to be side-by-side like on
>     sphinx-gallery.readthedocs.io? They're appearing one above the other.

> Hmmm not sure to be perfectly honest.

Depends on your screen size (actually on the size of the div that they
are in, which will vary from theme to theme.

I tend to prefer them side-by-side, unless the screen is small.

jnothman commented:
Hmmmm mine should be wide enough, but I agree in any case: merge if we know
no severe issue

On 17 Jan 2017 9:35 pm, "Gael Varoquaux" <notifications@github.com> wrote:

> > Are the download buttons meant to be side-by-side like on
> > sphinx-gallery.readthedocs.io? They're appearing one above the other.
>
> > Hmmm not sure to be perfectly honest.
>
> Depends on your screen size (actually on the size of the div that they
> are in, which will vary from theme to theme.
>
> I tend to prefer them side-by-side, unless the screen is small.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7986#issuecomment-273098329>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz681YMlRYFqOGBxfsAscegJEYzprJks5rTJlmgaJpZM4LE3zi>
> .
>

Titan-C commented:
> Are the download buttons meant to be side-by-side like on
> sphinx-gallery.readthedocs.io? They're appearing one above the other.

This is totally strange. It is a conflict in the CSS I think. If you
make the screen very narrow they are stacked. Then as you widen the
screen they place themselves side-by-side and as you continue enlarging
the window the get back to be stacked.

ogrisel commented:
+1 for merging to fix CI and fix the cosmetic link layout issue in a latter PR.

lesteve commented:
Merging this one, thanks a lot @Titan-C !

jnothman commented:
Great. Though we're going to continue to run into CircleCI failures in PRs
until we fix #8058.

On 18 January 2017 at 09:43, Loïc Estève <notifications@github.com> wrote:

> Merged #7986 <https://github.com/scikit-learn/scikit-learn/pull/7986>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/7986#event-926622385>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz6wF2le1mi52a1cUHTgGw2NKIKIsxks5rTUQegaJpZM4LE3zi>
> .
>
